common/app/common/dfp/PageskinAdAgent.scala

```diff
@@ -25,11 +25,25 @@ trait PageskinAdAgent {
       } else Seq.empty
     } else {
       val targetingMap = toMap(metaData.commercial.map(_.adTargeting(edition)).getOrElse(Set.empty))
-      val keywordTargeting = targetingMap.get("k") match {
-        case Some(values: MultipleValues) => values.values.toSeq
-        case _ => Seq.empty
-      }
-      candidates filter { sponsorship => sponsorship.keywords.intersect(keywordTargeting).nonEmpty }
+
+      // Little DRY helper
+      val targetingMapValues = ((map: Map[String, AdTargetParamValue], key: String) =>
```

Comment:
I think this is probably an over abstraction. You can simply do:

```
val keywordTargeting = targetingMap.get("k").map(_.values).getOrElse(Seq.empty)
val seriesTargeting = targetingMap.get("se").map(_.values).getOrElse(Seq.empty)
```

Reply:
Not sure it is equivalent to the case class match, as AdTargetParamValue is a trait that doesn't have `values`